### ISSUE TITLE
Implement GGN as an alternative Hessian approximation

### ISSUE THREAD

**Issue by MarkArdman** [navimakarov/LaplaceRedux.jl#5]
### User story:
- User must be able to use all of the existing functionality with GGN (Generalised Gauss-Newton approximation)

### Subtasks:
- [x] Compute GGN approximation using Jacobians

**Comment by severinbratus** [comment]
> Modify the backpropagation algorithm to compute the loss gradient and Jacobians

I believe this is inaccurate -- we do not ever have access to the backprop algorithm of the neural network, as we are constructing the Laplace approximation post hoc. Then even if this "backprop modification" does happen in the sense that Zygote.jl applies the chain rule to propagate the error back, and thus get all the gradients of the loss function wrt the weights, this should not be part of our issue, as the functionality is implemented in Zygote.
